rfcs/rdfa-editor-stage-1.md

```diff
@@ -0,0 +1,587 @@
+So we should be able to express in the plugin: "if the user is editing the object value of a triple where the predicate is `schema:email`, 
+provide some support.
+Ok simple enough, right? Calculate the triple which is currently being edited, and send that information to plugins.
+
+However, what if a plugin is not interested in email-addresses, but in Persons? That same address-book plugin, for example,
+could provide a way to add new entries by detecting information in the document about people it doesn't recognize, and
+rendering an "Add to contacts" button. 
+For that, calculating the abovementioned triple would not be enough. We would need to look further in the document and
+see that that same blankNode, of which we are currently editing the email address, also is the subject of the triple
+`<blankNode> a schema:Person`, or colloquially, that it is an instance of the Person class.
+
+And imagine even further, that this `blankNode` we are looking at, is also the _object_ of a triple. It could
+be for example, that this `Person` is the `besluit:chairman` of a resource which is an instance of a `besluit:Zitting` class.
+
+While editing that same email-adress, should a plugin interested in `Zitting`-type resources also be notified?
+And what about non-collapsed selections? What does "the triple being edited" mean if a selection covers nodes which are part of different triples?
+In the example above, imagine a selection starting in the middle of the phone number, and ending in the middle of the email-address.
+Should the email plugin trigger? Or the phone-number plugin?
+
+I am of the opinion that this question cannot be answered a priori. Therefore, it should be up to the plugin to decide, and up to the editor 
+to provide as much information to the plugin as possible, in a form that can answer the following questions easily:
+
+- is the selection currently in/around/touching an object value of a triple with given subject or predicate?
+- is the selection currently in/around/touching a something which is "related" 
+to a resource which is also the subject of a given triple? (most commonly, we'd be interested in the type of the resource)
+- does a range contain/touch a triple with a certain shape?
+- given a triple, which node defines its subject, which its predicate and which its object? (this also needs to support duplicate triples)
+- give me all nodes which are the subject of a given triple
+- give me all nodes which are the object of a given triple
+
+If this is starting to sound a lot like the questions you would ask a triplestore, that makes a lot of sense.
+In essence, the document _is_ a triplestore, just a very clumsy one, and one with added structure (the html markup).
+
+The idea here is to provide a single interface for all these questions. Whether the knowledge required for the questions is
+calculated ad-hoc or kept up-to-date throughout document edits is not specified here. Likely the former
```

Comment:
What about the initial load? I think keeping the store up to date could be expensive but not really impactful when changing the document (if you are just writing or deleting normally) but what I'm afraid is how much will take for long documents to load or when deleting/pasting large amounts of texts.
Anyway I think the datastore is a good idea, better than the context we had before

Reply:
yeah I don't know how the implementation will be, thats why this is just an interface.
I think initially we calculate things on the fly and see where we can gain performance by precalculating/caching